### ISSUE TITLE
record_summaries_every_n_global_steps() should not execute code unless global_step % n == 0

### ISSUE THREAD

**Issue by jpatts** [tensorflow/tensorflow#26225]
* Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
  No
* OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
  Ubuntu 18.04
* Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
  No
* TensorFlow installed from (source or binary):
  Source
* TensorFlow version (use command below):
  b'v1.12.0-5845-g764109a352' 1.12.0
* Python version:
  3.6.7
* Bazel version (if compiling from source):
  Invocation ID: 42251854-036f-415c-8a52-76aac8520ea0
  Build label: 0.21.0
  Build time: Wed Dec 19 12:58:44 2018 (1545224324)
* GCC/Compiler version (if compiling from source):
  gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
* CUDA/cuDNN version:
  Cuda compilation tools, release 10.0, V10.0.130
* GPU model and memory:
  GTX 1060 Max Q, 6gb VRAM

**Describe the current behavior**
`tf.contrib.summary.record_summaries_every_n_global_steps(n, global_step)` may not be recording summaries unless `global_step % n == 0`, HOWEVER, it is running the code within its block anyways. In my opinion this is a huge waste of resources, especially when image logging is being done and transformations are required. Furthermore, there is no information of this behaviour on the documentation. Finally, it defeats the purpose of the function.
https://www.tensorflow.org/api_docs/python/tf/contrib/summary/record_summaries_every_n_global_steps
**Describe the expected behavior**
It should not run the code within its "with" block when `global_step % n != 0`, or documentation should be added that describes the current behaviour. It defeats the purpose of it when I can combine an if statement with `tf.contrib.summary.always_record_summaries():` to produce better results.
**Code to reproduce the issue**
```
global_step = tf.train.get_or_create_global_step()
writer = tf.contrib.summary.create_file_writer('./test')
writer.set_as_default()

images = []
for i in range(100):
        images.append(np.random.uniform(0, 255, (32, 32, 3)).astype(np.float32))
images = np.array(images)
data = tf.data.Dataset.from_tensor_slices((images))

for i in range(10000):
        batch = data.batch(len(images))
        for image in batch:
                with tf.contrib.summary.record_summaries_every_n_global_steps(100, global_step):
                        print(global_step.numpy())
                        tf.contrib.summary.image('image', image)
        global_step.assign_add(1)
```
**Other info / logs**
```
0
1
2
3
4
5
6
....
9999
```
The images were still logged every global_step % n == 0 iterations, yet as can be seen here, the code within the block still executed every global_step. I shouldn't need to add an `if global_step % n == 0:` outside of the `with tf.contrib.summary.record_summaries_every_n_global_steps(n, global_step):` function. It literally defeats the purpose of it, as I might as well just use `with tf.contrib.summary.always_record_summaries():` inside of an `if global_step % n == 0:`.

**Comment by jvishnuvardhan** [comment]
@jpatts This is more related to tensorboard Repo. Please post the issue [here](https://github.com/tensorflow/tensorboard/issues) and the Tboard Team will take care of that issue. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will open a new issue. Thanks!
